File: packages/java/anip-core/src/main/java/dev/anip/core/BindingRequirement.java
package dev.anip.core;

/**
 * Declares that a capability requires a bound value.
 */
public class BindingRequirement {

    private String type;
    private String field;
    private String sourceCapability;
    private String maxAge;

    /** No-arg constructor for Jackson deserialization. */
    public BindingRequirement() {}

    public BindingRequirement(String type, String field, String sourceCapability, String maxAge) {
        this.type = type;
        this.field = field;
        this.sourceCapability = sourceCapability;
        this.maxAge = maxAge;
    }

    public String getType() {
        return type;
    }

    public String getField() {
        return field;
    }

    public String getSourceCapability() {
        return sourceCapability;
    }

    public String getMaxAge() {
        return maxAge;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setField(String field) {
        this.field = field;
    }

    public void setSourceCapability(String sourceCapability) {
        this.sourceCapability = sourceCapability;
    }

    public void setMaxAge(String maxAge) {
        this.maxAge = maxAge;
    }
}

File: packages/csharp/src/Anip.Core/BindingRequirement.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class BindingRequirement
{
    [JsonPropertyName("type")]
    public string Type { get; set; } = "";

    [JsonPropertyName("field")]
    public string Field { get; set; } = "";

    [JsonPropertyName("source_capability")]
    public string? SourceCapability { get; set; }

    [JsonPropertyName("max_age")]
    public string? MaxAge { get; set; }
}
